source/state_representation/src/robot/JointState.cpp

```diff
@@ -67,6 +67,66 @@ JointState JointState::Random(const std::string& robot_name, const std::vector<s
   return random;
 }
 
+unsigned int JointState::get_joint_index(const std::string& joint_name) const {
+  auto finder = std::find(this->names_.begin(), this->names_.end(), joint_name);
+  if (finder == this->names_.end()) {
+    throw JointNotFoundException("The joint with name '" + joint_name + "' could not be found in the joint state.");
+  }
+  return std::distance(this->names_.begin(), finder);
+}
+
+double JointState::get_position(const std::string& joint_name) const {
+  return this->positions_(this->get_joint_index(joint_name));
+}
+
+double JointState::get_position(unsigned int joint_index) const {
+  if (joint_index > this->get_size()) {
+    throw JointNotFoundException(
+        "Index '" + std::to_string(joint_index) + "' is out of range for joint state with size"
+            + std::to_string(this->get_size()));
+  }
```

Comment:
there's a bit of code duplication in these four methods. could you define a local static method `assert_index_in_range` or similar that takes the index and size and does this check?